oscar/cdk/stacks/lambda_stack.py

```diff
@@ -0,0 +1,260 @@
+                    "bedrock:ListFoundationModels",
+                    "bedrock:GetKnowledgeBase",
+                    "bedrock:ListKnowledgeBases",
+                    "bedrock:GetInferenceProfile",
+                    "bedrock:ListInferenceProfiles",
+                    "bedrock-agent-runtime:Retrieve",
+                    "bedrock-agent-runtime:RetrieveAndGenerate",
+                    "bedrock-agent-runtime:InvokeAgent"
+                ],
+                resources=["*"]
+            )
+        )
+
+        # Add permissions for DynamoDB with least privilege
+        role.add_to_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "dynamodb:GetItem",
+                    "dynamodb:PutItem",
+                    "dynamodb:UpdateItem",
+                    "dynamodb:DeleteItem",
+                    "dynamodb:Query"
+                ],
+                resources=[
+                    sessions_table.table_arn,
+                    context_table.table_arn
+                ]
+            )
+        )
+        
+        # Add permissions for Lambda to invoke itself asynchronously
+        function_name = os.environ.get("LAMBDA_FUNCTION_NAME", "oscar-slack-bot")
+        role.add_to_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "lambda:InvokeFunction"
+                ],
+                resources=[
+                    f"arn:aws:lambda:*:*:function:{function_name}"
+                ]
+            )
+        )
+        
+        return role
+    
+    def _create_lambda_function(self) -> lambda_.Function:
+        """
+        Create the Lambda function with placeholder code.
+        
+        Returns:
+            The created Lambda function
+        """
+        function_name = os.environ.get("LAMBDA_FUNCTION_NAME", "oscar-slack-bot")
+        
+        return lambda_.Function(
+            self, "OscarSlackBotFunction",
+            function_name=function_name,
+            runtime=lambda_.Runtime.PYTHON_3_9,
+            handler="app.lambda_handler",
+            code=lambda_.Code.from_inline("""
+import os
+def lambda_handler(event, context):
+    return {
+        'statusCode': 200,
+        'body': 'Lambda function deployed successfully. Will be updated with full code.'
+    }
+"""),
+            timeout=Duration.seconds(30),
+            memory_size=512,
+            environment=self._get_lambda_environment_variables(),
+            role=self.lambda_role
+        )
+    
+    def _create_api_gateway(self) -> apigateway.LambdaRestApi:
+        """
+        Create the API Gateway for the Lambda function.
+        
+        Returns:
+            The created API Gateway
+        """
+        api = apigateway.LambdaRestApi(
+            self, "OscarSlackBotApi",
+            handler=self.lambda_function,
+            proxy=False,
+            default_cors_preflight_options=apigateway.CorsOptions(
+                allow_origins=apigateway.Cors.ALL_ORIGINS,
```

Comment:
Can add a TODO to improve this so that only specific origin is used based on user config.